prow/plugins/transfer-issue/transfer-issue.go

```diff
@@ -0,0 +1,149 @@
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package transferissue implements the `/transfer-issue` command which allows members of the org
+// to transfer issues between repos
+package transferissue
+
+import (
+	"context"
+	"fmt"
+	"regexp"
+
+	githubql "github.com/shurcooL/githubv4"
+	"github.com/sirupsen/logrus"
+
+	"k8s.io/test-infra/prow/config"
+	"k8s.io/test-infra/prow/github"
+	"k8s.io/test-infra/prow/pluginhelp"
+	"k8s.io/test-infra/prow/plugins"
+)
+
+const pluginName = "transfer-issue"
+
+var (
+	transferRe = regexp.MustCompile(`(?mi)^/transfer(?:-issue)?\s+(.+)$`)
+)
+
+type githubClient interface {
+	GetRepo(org, name string) (github.FullRepo, error)
+	CreateComment(org, repo string, number int, comment string) error
+	IsMember(org, user string) (bool, error)
+	MutateWithGitHubAppsSupport(context.Context, interface{}, githubql.Input, map[string]interface{}, string) error
+}
+
+func init() {
+	plugins.RegisterGenericCommentHandler(pluginName, handleGenericComment, helpProvider)
+}
+
+func helpProvider(_ *plugins.Configuration, _ []config.OrgRepo) (*pluginhelp.PluginHelp, error) {
+	pluginHelp := &pluginhelp.PluginHelp{
+		Description: "The transfer-issue plugin transfers a GitHub issue from one repo to another in the same organization.",
+	}
+	pluginHelp.AddCommand(pluginhelp.Command{
+		Usage:       "/transfer[-issue] <destination repo in same org>",
+		Description: "Transfers an issue to a different repo in the same org.",
+		Featured:    true,
+		WhoCanUse:   "Org members.",
+		Examples:    []string{"/transfer-issue kubectl", "/transfer test-infra"},
+	})
+	return pluginHelp, nil
+}
+
+func handleGenericComment(pc plugins.Agent, e github.GenericCommentEvent) error {
+	return handleTransfer(pc.GitHubClient, pc.Logger, e)
+}
+
+func handleTransfer(gc githubClient, log *logrus.Entry, e github.GenericCommentEvent) error {
+	org := e.Repo.Owner.Login
+	srcRepoName := e.Repo.Name
+	srcRepoPair := org + "/" + srcRepoName
+	user := e.User.Login
+
+	if e.IsPR || e.Action != github.GenericCommentActionCreated {
+		return nil
+	}
+	matches := transferRe.FindAllStringSubmatch(e.Body, -1)
+	if len(matches) == 0 {
+		return nil
+	}
+	if len(matches) != 1 || len(matches[0]) != 2 {
+		return gc.CreateComment(
+			org, srcRepoName, e.Number,
+			plugins.FormatResponseRaw(e.Body, e.HTMLURL, user, "/transfer-issue must only be used once and with a single destination repo."),
+		)
+	}
+
+	dstRepoName := matches[0][1]
+	dstRepoPair := org + "/" + dstRepoName
+
+	dstRepo, err := gc.GetRepo(org, dstRepoName)
+	if err != nil {
+		log.WithError(err).Errorf("could not fetch destination repo: %s", dstRepoPair)
```

Comment:
We generally prefer structured logs (`WithField`/`WithFields`) over `Errorf`. Also, this is processing user input, so IMO ti warrants a warning at best, otherwise logs would be spammed by typos etc.

Reply:
I'll switch it to a warning since `GetRepo`'s failure doesn't differentiate be a not found and a failed request.